packages/@vue/cli/lib/invoke.js

```diff
@@ -132,9 +135,18 @@ async function runGenerator (context, plugin, pkg = getPkg(context)) {
     await pm.install()
   }
 
-  if (createCompleteCbs.length) {
+  if (afterInvokeCbs.length) {
     logWithSpinner('⚓', `Running completion hooks...`)
-    for (const cb of createCompleteCbs) {
+    for (const cb of afterInvokeCbs) {
+      await cb()
+    }
+    stopSpinner()
+    log()
+  }
+
+  if (afterAnyInvokeCbs.length) {
+    logWithSpinner('⚓', `Running completion hooks from other plugins...`)
```

Comment:
Does this include the `afterAnyInvoke` hook from the currently running generator?

Reply:
Yes, it does.

Reply:
From all the current plugins.

Reply:
But the order of `afterAnyInvoke` callbacks is first all other plugins' hook, then the current plugin's `afterAnyInvoke` hook.

The exact result depends on which plugin did you invoke first.
This looks confusing to me.
It is the same kind of non-determinism like `npm install`. We should avoid it.

Reply:
Also the log is misleading in that case.

Reply:
Fixed this. Current plugins will run first and then others. The order is now documented.

Reply:
It no longer triggers other plugins' `afterAnyInvoke` hook as I tested.

I'm also a little bit skeptical about this approach - this is still non-deterministic IMO.
For example, if a user has two plugins with `afterAnyInvoke` hook, say, `eslint` and `typescript`, the order will be:

- `eslint` -> `typescript` on creation
- `eslint` -> `typescript` if user add `eslint` first
- `typescript` -> `eslint` if user add `typescript` first

My idea is to keep the order of `afterAnyInvoke` in sync with that of plugin loading (for now, it is the key enumeration order in `package.json`) and then figure out a way to specify plugin order in a separate PR. What do you think?

Reply:
Good idea. I didn't think of it like that. Made it completely deterministic like how you said. Should be working now.